### PR TITLE
Added solution of 0102

### PR DESCRIPTION
**Pull request by Srijita-Mandal**
## Change Summary

I added the solution of the leetcode problem 0102

## Checklist

**If you haven't fulfilled the below requirements, your PR won't be reviewed.**

### General 

- [x] Make sure you've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md) 
- [x] Make sure you've followed the template format
- [x] Make sure you've started the app locally and verified all the content and all links (if applicable) must be accessible correctly
- [x] Make sure you've applied LaTex for variables or formulas
- [x] Make sure you've formatted your code well (Code Block is rendered using `<Tabs/>`). 
      Below is a C++ example. You can include multiple langs using `<TabItem/>` if necessary.
      
````
<Tabs>

<TabItem value="cpp" label="C++">
<SolutionAuthor name="@YOUR_ALIAS"/>

```cpp
// Your code goes here
// Prefer K&R Coding style
```

</TabItem>

</Tabs>
````
- [x] Make sure you've explained your code well (write comments above each line)
- [x] Make sure you've written your explanation well and it is easy to understand for beginners
- [x] Make sure you've included Time & Space Complexity Analysis 

### For Tutorial Articles

- [ ] Make sure you've explained your topic using 1 - 3 LC problems rather than using external problems. Solutions must be taken with explanations as well.
- [ ] Make sure you've provided suggested problems at the end using the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [ ] Make sure you've given credits / references if you use external resources. For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).
- [ ] Make sure you've included a meaningful approach name

### CODE REVIEW

**Review by wingkwong — CHANGES_REQUESTED** [review]
Rejected. Please look at the checklist again.

**Comment by Srijita-Mandal** [comment]
@wingkwong I have checked the list

File: solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            }
+            else{
+                v.push_back(cur->val);
+                if(cur->left!=NULL) q.push(cur->left);
+                if(cur->right!=NULL) q.push(cur->right);
+            }
+        }
+        return ans;
+    }
+};
```
Comment (wingkwong) [inline solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md:78]: 1. the code format is already wrong. See General point 5.
2. comments should be above each line. not inline. See General point 6.

File: solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+*  The number of nodes in the tree is in the range `[0, 2000]`.
+*  `-1000 <= Node.val <= 1000`
+
+
+## Approach 1: Using Queue Data Structure
+
+- Take one 2D vector `ans` to return the final tree-traversal, and another vector `v` to store seperate levels.
+- Take a queue, store `root` node inside it and a `NULL` node for seperating levels.
+- Until the queue is not empty, pop one by one nodes from queue. if the node is `NULL` and queue is not empty, push the `NULL` again inside the queue. Push `v` into `ans` as one level is completed and clear `v`.
+- If the node is not null, push the value into `v` and push its left and right nodes into queue (if not null).
```
Comment (wingkwong) [inline solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md:46]: use LaTex, not backtick

File: solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+**Time Complexity:**  O(N)
```
Comment (wingkwong) [inline solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md:48]: 1. Apply LaTex. See General point 4.
2. Missing Space Complexity. See General point 8.

**Review by wingkwong — CHANGES_REQUESTED** [review]
I suggest you to read the checklist again. Not just tick them without checking.

Besides, you deleted my `package-lock.json`. don't commit that.

**Comment by Srijita-Mandal** [comment]
@wingkwong I have made the changes. Please check if it is okay now.

File: package-lock.json [hunk 4]
```diff
@@ -2509,6 +2509,28 @@
         "node": ">=6.0.0"
       }
     },
+    "node_modules/@jridgewell/source-map": {
```
Comment (vigneshshiv) [inline package-lock.json:2512]: Why there's a change in package-lock.json file ? 
These files shouldn't be added for PR.

**Review by vigneshshiv — CHANGES_REQUESTED** [review]
package-lock.json shouldn't be added to LC solutions PR. Undo the changes

File: solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+                v.push_back(cur->val);
+
+                if(cur->left!=NULL) q.push(cur->left);
+
+                if(cur->right!=NULL) q.push(cur->right);
+            }
+        }
+        return ans;
+    }
+};
```
Comment (wingkwong) [inline solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md:98]: Prefer K&R style

https://en.wikipedia.org/wiki/Indentation_style#K&R_style

**Review by wingkwong — CHANGES_REQUESTED** [review]
don't commit package-lock.json

**Comment by Srijita-Mandal** [comment]
Changed the code indentation to K&R style. And didn't commit `package-lock.json` this time. I didn't really made any changes in the `json` file. Can't understand why this is changed. Please check if this is okay this time, if not, please tell me how to resolve it.

**Review by wingkwong — CHANGES_REQUESTED** [review]
As long as I see this package-lock.json in Files changed, I reject.

image

**Comment by vigneshshiv** [comment]
@Srijita-Mandal You need to reset your changes based on the last git commit (previous the this). Once you've done reset, your this changes will be unstaged. The undo the json changes and then commit only the relavant changes.

**Comment by Srijita-Mandal** [comment]
I have made a new PR #225  regarding this issue, with preferable changes. Please review it.

**Comment by wingkwong** [comment]
Closing this one. Move to PR225.

**Pull request closed without merging**